addons/interaction/functions/fnc_switchWeaponAttachment.sqf

```diff
@@ -38,22 +46,42 @@ if (_removeOld && {!([_unit, _oldAttachment] call CBA_fnc_canAddItem)}) exitWith
     };
 };
 
+[_unit, "Gear"] call EFUNC(common,doGesture);
+
 if (_removeOld) then {
     [{
-        params ["_unit", "_weapon", "_oldAttachment"];
-        switch (_weapon) do {
-            case (primaryWeapon _unit): {_unit removePrimaryWeaponItem _oldAttachment;};
-            case (handgunWeapon _unit): {_unit removeHandgunItem _oldAttachment;};
-            default {_unit removeSecondaryWeaponItem _oldAttachment;};
+        params ["_unit", "_oldAttachment", "", "_currWeaponType"];
+
+        switch (_currWeaponType) do {
+            case 0: {_unit removePrimaryWeaponItem _oldAttachment};
+            case 1: {_unit removeHandgunItem _oldAttachment};
+            case 2: {_unit removeSecondaryWeaponItem _oldAttachment};
+            default {};
         };
+
         _unit addItem _oldAttachment;
-    }, [_unit, _weapon, _oldAttachment], 0.3] call CBA_fnc_waitAndExecute;
+
+        // If adding a new item, let that code handle raising EH
+        // Delete addNew from array, to be able to pass _this to EH
+        if (_this deleteAt 4) exitWith {};
+
+        ["CBA_attachmentSwitched", _this] call CBA_fnc_localEvent;
+    }, [_unit, _oldAttachment, _newAttachment, _currWeaponType, _addNew], 0.3] call CBA_fnc_waitAndExecute;
 };
 
 if (!_addNew) exitWith {};
 
 [{
-    params ["_unit", "_weapon", "_newAttachment"];
-    _unit addWeaponItem [_weapon, _newAttachment];
+    params ["_unit", "", "_newAttachment"];
+
+    // Delete weapon from array, to be able to pass _this to EH
+    _unit addWeaponItem [_this deleteAt 4, _newAttachment];
+
     [[getText (configFile >> "CfgWeapons" >> _newAttachment >> "picture"), 4], true] call CBA_fnc_notify;
-}, [_unit, _weapon, _newAttachment], 1] call CBA_fnc_waitAndExecute;
+
+    if (_unit == ACE_player) then {
+        playSound "click";
```

Comment:
I'd expect that although the sound couldn't possibly be very loud by any stretch, someone like 0.5 meters, maybe 1 meter, away should be able to hear it in a mostly quiet environment.

Reply:
All mode switching sounds are local only (both in CBA and ACE), so I won't be changing those now - but maybe in the future.